api-registration/server/index.js

```diff
@@ -0,0 +1,74 @@
+require('dotenv/config');
+const pg = require('pg');
+const argon2 = require('argon2'); // eslint-disable-line
+const express = require('express');
+const ClientError = require('./client-error');
+const errorMiddleware = require('./error-middleware');
+
+const db = new pg.Pool({ // eslint-disable-line
+  connectionString: process.env.DATABASE_URL,
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+const app = express();
+
+const jsonMiddleware = express.json();
+
+app.use(jsonMiddleware);
+
+app.post('/api/auth/sign-up', (req, res, next) => {
+  const { username, password } = req.body;
+  if (!username || !password) {
+    throw new ClientError(400, 'username and password are required fields');
+  }
+
+  /* your code starts here */
+  // const argon2 = require('argon2');
+
+  // const password = process.argv[2];
+
+  argon2
+    .hash(password)
+    .then(hashedPassword => {
+      // console.log(hashedPassword);
+      const params = [username, hashedPassword];
+      const sql = `
+      insert into "users" ("username", "hashedPassword")
+      values ($1, $2)
+      returning "UserId", "username", "createdAt"
+      `;
+
+      db.query(sql, params)
+        .then(result => {
+          // console.log('test', result.rows);
+          // // const [newInfo] = result.row;
+          res.status(201).json(result.row);
```

Comment:
I'm not sure this is actually sending back the desired data in the response body.

Reply:
Unless node-postgres has a Result.row property I didn't know about

Reply:
<img width="576" alt="Screen Shot 2022-04-05 at 5 19 18 PM" src="https://user-images.githubusercontent.com/40590740/161871695-df3fc987-7db7-4688-986e-3d8b06da63a3.png">


Fixed result.row as well as returning "userId"